### PR TITLE
#403 new dhc pipe integration into dev

### PR DESCRIPTION
**Pull request by MichaMans**
This adds the more universal DHC pipe for DHC simulations

### CODE REVIEW

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    annotation (Line(points={{-50,-70},{-40,-70}}, color={0,127,255}));
+  connect(senTemIn3.port_b, pip3.port_a)
+    annotation (Line(points={{-20,-70},{-10,-70}}, color={0,127,255}));
+  annotation (
+    __Dymola_Commands(file="modelica://AixLib/Resources/Scripts/Dymola/Fluid/DistrictHeatingCooling/Pipes/Examples/PlugFlowPipeZeta.mos"
+                      "Simulate and plot"),
+    experiment(StopTime=1000, Tolerance=1e-006),
+    Documentation(info="<html><p>
+  Basic test of model <a href=
+  \"modelica://AixLib.Fluid.DistrictHeatingCooling.Pipes.PlugFlowPipeZeta\">
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:256]:
```suggestion
  \"modelica://AixLib.Fluid.DistrictHeatingCooling.Pipes.DHCPipe\">
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+  AixLib.Fluid.DistrictHeatingCooling.Pipes.PlugFlowPipeZeta</a>. This
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:257]:
```suggestion
  AixLib.Fluid.DistrictHeatingCooling.Pipes.DHCPipe</a>. This
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  test includes an inlet temperature step under a constant mass flow
+  rate.
+</p>
+<ul>
+  <li>September 25, 2019 by Nils Neuland:<br/>
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:262]:
```suggestion
   <li>November 12, 2020, by Michael Mans:<br/>
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+  connect(pip3.ports_b[1], senTemOut3.port_a)
+    annotation (Line(points={{10,-70},{20,-70}}, color={0,127,255}));
+  connect(senTemOut3.port_b, sin3.ports[1])
+    annotation (Line(points={{40,-70},{52,-70}}, color={0,127,255}));
+  connect(sou3.ports[1], senTemIn3.port_a)
+    annotation (Line(points={{-50,-70},{-40,-70}}, color={0,127,255}));
+  connect(senTemIn3.port_b, pip3.port_a)
+    annotation (Line(points={{-20,-70},{-10,-70}}, color={0,127,255}));
+  annotation (
+    __Dymola_Commands(file="modelica://AixLib/Resources/Scripts/Dymola/Fluid/DistrictHeatingCooling/Pipes/Examples/PlugFlowPipeZeta.mos"
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:251]:
```suggestion
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+                      "Simulate and plot"),
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:252]:
```suggestion
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+    experiment(StopTime=1000, Tolerance=1e-006),
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:253]:
```suggestion
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 7]
```diff
@@ -0,0 +1,267 @@
+within AixLib.Fluid.DistrictHeatingCooling.Pipes.Examples;
+model DHCPipe "Simple example of PlugFlowPipe"
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:2]:
```suggestion
model DHCPipe "Simple example of DHCPipe and its four different modes"
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+  annotation (
+    __Dymola_Commands(file="modelica://AixLib/Resources/Scripts/Dymola/Fluid/DistrictHeatingCooling/Pipes/Examples/PlugFlowPipeZeta.mos"
+                      "Simulate and plot"),
+    experiment(StopTime=1000, Tolerance=1e-006),
+    Documentation(info="<html><p>
+  Basic test of model <a href=
+  \"modelica://AixLib.Fluid.DistrictHeatingCooling.Pipes.PlugFlowPipeZeta\">
+  AixLib.Fluid.DistrictHeatingCooling.Pipes.PlugFlowPipeZeta</a>. This
+  test includes an inlet temperature step under a constant mass flow
+  rate.
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/Examples/DHCPipe.mo:259]:
```suggestion
  rate. All four possible applications of the DHC pipe model are shown as examples.
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo [hunk 9]
```diff
@@ -0,0 +1,496 @@
...
+    final nPorts=nPorts + 1,
+    final T_start=T_start_out,
+    final energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    final mSenFac = if rho_default > 500 then 1 else 10)
+    "Control volume connected to ports_b. Represents equivalent pipe wall thermal capacity."
+    annotation (Placement(transformation(extent={{60,20},{80,40}})));
+
+  AixLib.Utilities.HeatTransfer.CylindricHeatTransfer cylHeaTra1(
+    final energyDynamics=energyDynamics,
+    final rho=rhoSoi,
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo:191]:
```suggestion
    final rho=rho_soi,
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo [hunk 10]
```diff
@@ -0,0 +1,496 @@
...
+    final d_in=dh + 2*thickness,
+    final d_out=d_in + thickness_ground/3,
+    final length=length,
+    final lambda=lambda,
+    T0=283.15) if use_soil
+    annotation (Placement(transformation(extent={{-10,20},{10,40}})));
+
+  AixLib.Utilities.HeatTransfer.CylindricHeatTransfer cylHeaTra2(
+    final energyDynamics=energyDynamics,
+    final rho=rhoSoi,
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo:202]:
```suggestion
    final rho=rho_soi,
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo [hunk 11]
```diff
@@ -0,0 +1,496 @@
...
+    final c=c,
+    final d_in=dh + 2*thickness + thickness_ground/3,
+    final d_out=d_in + 2*thickness_ground/3,
+    final length=length,
+    final lambda=lambda,
+    T0=283.15) if use_soil
+    annotation (Placement(transformation(extent={{-10,46},{10,66}})));
+  AixLib.Utilities.HeatTransfer.CylindricHeatTransfer cylHeaTra3(
+    final energyDynamics=energyDynamics,
+    final rho=rhoSoi,
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo:212]:
```suggestion
    final rho=rho_soi,
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo [hunk 12]
```diff
@@ -0,0 +1,496 @@
...
+    final mSenFac = if rho_default > 500 then 1 else 10)
+    "Control volume connected to ports_b. Represents equivalent pipe wall thermal capacity."
+    annotation (Placement(transformation(extent={{60,20},{80,40}})));
+
+  AixLib.Utilities.HeatTransfer.CylindricHeatTransfer cylHeaTra1(
+    final energyDynamics=energyDynamics,
+    final rho=rhoSoi,
+    final c=c,
+    final d_in=dh + 2*thickness,
+    final d_out=d_in + thickness_ground/3,
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo:194]:
```suggestion
    final d_out=d_in + thickness_soi/3,
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo [hunk 13]
```diff
@@ -0,0 +1,496 @@
...
+    final length=length,
+    final lambda=lambda,
+    T0=283.15) if use_soil
+    annotation (Placement(transformation(extent={{-10,20},{10,40}})));
+
+  AixLib.Utilities.HeatTransfer.CylindricHeatTransfer cylHeaTra2(
+    final energyDynamics=energyDynamics,
+    final rho=rhoSoi,
+    final c=c,
+    final d_in=dh + 2*thickness + thickness_ground/3,
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo:204]:
```suggestion
    final d_in=dh + 2*thickness + thickness_soi/3,
```

File: AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo [hunk 14]
```diff
@@ -0,0 +1,496 @@
...
+    final d_out=d_in + 2*thickness_ground/3,
+    final length=length,
+    final lambda=lambda,
+    T0=283.15) if use_soil
+    annotation (Placement(transformation(extent={{-10,46},{10,66}})));
+  AixLib.Utilities.HeatTransfer.CylindricHeatTransfer cylHeaTra3(
+    final energyDynamics=energyDynamics,
+    final rho=rhoSoi,
+    final c=c,
+    final d_in=dh + 2*thickness + 2*thickness_ground/3,
```
Comment (TobiasBlacha) [inline AixLib/Fluid/DistrictHeatingCooling/Pipes/DHCPipe.mo:214]:
```suggestion
    final d_in=dh + 2*thickness + 2*thickness_soi/3,
```

**Review by TobiasBlacha — CHANGES_REQUESTED** [review]

**Review by TobiasBlacha — APPROVED** [review]

**Comment by MichaMans** [comment]
@TobiasBlacha could you have a quick look again and maybe approve it? Checks are now passing

**Review by TobiasBlacha — APPROVED** [review]

**Pull request merged**